### PR TITLE
Intelligently enable repo sets

### PR DESCRIPTION
**Pull request by zmc**

### CODE REVIEW

File: roles/common/tasks/main.yml [hunk 1]
```diff
@@ -7,6 +7,8 @@
     - empty.yml
   tags:
     - vars
+    # We need these vars for the entitlements tag to work
+    - entitlements
```
Comment (andrewschoen) [inline roles/common/tasks/main.yml]: Use the tag 'always'.  This is a magic ansible tag that ensures this will run whenever --tags is used.  I've done this in other places where vars are loaded as well.

If you do that, the 'vars' tag is unneeded.

File: roles/common/tasks/rhel-entitlements.yml [hunk 2]
```diff
@@ -14,26 +14,54 @@
...
+  when: subscription is defined
+
 - name: Register with subscription-manager.
   command: subscription-manager register
            --activationkey={{ subscription_manager_activationkey }}
            --org={{ subscription_manager_org }}
   no_log: true
-  when: subscription is defined and
-        subscription.rc != 0
+  when: rhsm_registered == "False"
```
Comment (andrewschoen) [inline roles/common/tasks/rhel-entitlements.yml]: Will this error if rhsm_registered is not defined?  I think it would.

File: roles/common/tasks/rhel-entitlements.yml [hunk 3]
```diff
@@ -14,26 +14,54 @@
...
+
+- name: Set replace_repos if newly-subscribed
+  set_fact:
+    replace_repos: True
   when: entitled|changed
 
-- name: enable necessary rhsm repos
+- name: Disable all rhsm repos
+  command: subscription-manager repos --disable '*'
+  when: replace_repos == "True"
```
Comment (andrewschoen) [inline roles/common/tasks/rhel-entitlements.yml]: Do we need to check to see if replace_repos is defined?  Or maybe we could set a defaults for these sorts of flags instead of always checking to make sure they're defined.

File: roles/common/tasks/rhel-entitlements.yml (hunk 2 above)
Comment (zmc) [inline roles/common/tasks/rhel-entitlements.yml]: Sure, but why would it not be defined?

File: roles/common/tasks/rhel-entitlements.yml (hunk 3 above)
Comment (zmc) [inline roles/common/tasks/rhel-entitlements.yml]: It will be defined, though.

File: roles/common/tasks/main.yml (hunk 1 above)
Comment (zmc) [inline roles/common/tasks/main.yml]: I'm confused.

I'm adding the entitlements tag so I can entitle machines as a separate step if I need to. I don't necessarily want this to run *always*. The vars tag was added in 5eece205, by you.

File: roles/common/tasks/rhel-entitlements.yml (hunk 2 above)
Comment (andrewschoen) [inline roles/common/tasks/rhel-entitlements.yml]: I think I was confused because the set_fact above has a ``when: subscription is defined`` which led me to believe that rhsm_registered might not be created in some circumstances.

File: roles/common/tasks/rhel-entitlements.yml [hunk 4]
```diff
@@ -14,26 +14,54 @@
   changed_when: false
   no_log: true
 
+- name: Set rhsm_registered
+  set_fact:
+    rhsm_registered: "{{ subscription.rc == 0 }}"
+  when: subscription is defined
```
Comment (andrewschoen) [inline roles/common/tasks/rhel-entitlements.yml]: I believe this is not needed as ``subscription`` looks to always be defined, right?

File: roles/common/tasks/rhel-entitlements.yml (hunk 3 above)
Comment (andrewschoen) [inline roles/common/tasks/rhel-entitlements.yml]: Only when ``entitled|changed``. What if we run this against a node that is already registered?  I would guess that ``replace_repos`` would not be defined. Might be wrong though.

File: roles/common/tasks/main.yml (hunk 1 above)
Comment (andrewschoen) [inline roles/common/tasks/main.yml]: Yeah, I quit using the 'vars' tag because I always forgot to include it when using --tags for tasks that required vars to be loaded.  Instead, I've started using ``always`` so I don't have to remember or specially tag include_vars blocks.

Just a suggestion, I find it easier this way.

http://docs.ansible.com/playbooks_tags.html#special-tags

File: roles/common/tasks/rhel-entitlements.yml (hunk 3 above)
Comment (zmc) [inline roles/common/tasks/rhel-entitlements.yml]: That's not quite true; it is set if ``repo_list is defined`` right above that.

File: roles/common/tasks/rhel-entitlements.yml (hunk 4 above)
Comment (zmc) [inline roles/common/tasks/rhel-entitlements.yml]: True, I think that line can safely be removed.

File: roles/common/tasks/rhel-entitlements.yml (hunk 2 above)
Comment (zmc) [inline roles/common/tasks/rhel-entitlements.yml]: Gotcha!

File: roles/common/tasks/main.yml (hunk 1 above)
Comment (zmc) [inline roles/common/tasks/main.yml]: I think if we want to switch from using the vars tag to the always tag, we should do that in a separate PR.

Comment (andrewschoen) [inline roles/common/tasks/main.yml]: Alright, do you think this is good to merge then?

Comment (zmc) [inline roles/common/tasks/main.yml]: I do
